### ISSUE TITLE
"Failed to create client: An unexpected D-Bus error occurred"

### ISSUE THREAD

**Issue by Epicanis** [eworm-de/pacredir#1]
Getting this on a couple of machines from pacredir. pacserve and pacdbserve both appear to successfully start and drop privileges, but pacredir fails with this cryptic error.

I can manually run "/usr/bin/pacredir -v" as a regular user, but via sudo or as the actual root login I get the same error, which presumably is why systemctl can't start it either.

```
# pacredir -v
pacredir: pacredir v0.3.3 arch/x86_64 (compiled: Dec 13 2017, 20:25:10)
Running as root, meh! Dropping privileges.
Using IPv4 only
Failed to create client: An unexpected D-Bus error occurred
```

Not sure what's going on... it *is* working correctly on some of my other systems, but two of them refuse with this error.

**Comment by eworm-de** [comment]
I guess that is caused by pending changes for `/etc/nsswitch.conf`. Make sure that lines starting with `passwd:` and `group:` end with `systemd`.

`pacredir` uses dynamic users, which requires the `systemd` module to resolve user names.

**Comment by Epicanis** [comment]
Ah, thanks - it was odd that pacserve and pacdbserve had not trouble with this but pacredir didn't. Updating nsswitch as specified and restarting the dbus service seems to have gotten pacredir working.

Thanks again, this is very handy considering how many arch machines I've got running here!

**Comment by xuanruiqi** [comment]
Old issue, but I'm getting it again, recently, but only when running as a systemd service.
I can confirm there's nothing wrong with my `nsswitch.conf`.

**Comment by eworm-de** [comment]
I've stumbled on this myself. For whatever reason a service can not connect to `avahi` via `dbus` when using dynamic user. Looks like this was introduced with `dbus` in version `1.14.0`... I did not yet investigate, though.

Currently I can give you two workarounds:
* Do not use dynamic user, this can be configured with simple drop-in snippet for the service.
* Switch to `dbus-broker`, which does not suffer the issue.

I am happy with the second, as I use it anyway. 😜
